auth-table-based-service/src/main/java/io/stargate/auth/table/AuthnTableBasedService.java

```diff
@@ -230,54 +237,64 @@ public AuthenticationSubject validateToken(String token) throws UnauthorizedExce
       throw new UnauthorizedException("authorization failed - missing token");
     }
 
-    UUID uuid;
     try {
-      uuid = UUID.fromString(token);
-    } catch (IllegalArgumentException exception) {
-      throw new UnauthorizedException("authorization failed - bad token");
-    }
-
-    String username;
-    try {
-      ResultSet resultSet =
-          dataStore
-              .queryBuilder()
-              .select()
-              .star()
-              .from(AUTH_KEYSPACE, AUTH_TABLE)
-              .where("auth_token", Predicate.EQ, uuid)
-              .build()
-              .execute(ConsistencyLevel.LOCAL_QUORUM)
-              .get();
-
-      if (resultSet.hasNoMoreFetchedRows()) {
-        throw new UnauthorizedException("authorization failed");
-      }
-
-      Row row = resultSet.one();
-      if (row.isNull("username")) {
-        throw new RuntimeException("unable to get username from token table");
-      }
-
-      int timestamp = row.getInt("created_timestamp");
-      username = row.getString("username");
-
-      dataStore
-          .queryBuilder()
-          .update(AUTH_KEYSPACE, AUTH_TABLE)
-          .ttl(tokenTTL)
-          .value("username", username)
-          .value("created_timestamp", timestamp)
-          .where("auth_token", Predicate.EQ, UUID.fromString(token))
-          .build()
-          .execute(ConsistencyLevel.LOCAL_QUORUM)
-          .get();
-    } catch (InterruptedException | ExecutionException e) {
-      logger.error("Failed to validate token", e);
+      return (AuthenticationSubject)
+          tokenCache.get(
+              token,
+              (Callable<AuthenticationSubject>)
+                  () -> {
+                    UUID uuid;
+                    try {
+                      uuid = UUID.fromString(token);
```

Comment:
Is this conversion really necessary or is it there only to confirm token is UUID? Should we use regex matching better?